tracks/julia/exercises/robot-name/mentoring.md

```diff
@@ -37,57 +37,60 @@ There's some basic stats you can do:
 
 using Random: shuffle!
 
-# Use Int32 to save some bits because this is quite a big vector.
+# We shuffle all possible robot names in advance so that we can issue them in
+# constant time.
+#
+# Robot names are expressed as integers because there's quite a lot of them and
+# this saves over 100 bits per name compared to storing a pointer (64 bits) to
+# a 5 character string (40 bits + length metadata).
 const names = shuffle!(Int32[0:26^2 * 10^3 - 1;])
 
 mutable struct Robot
-    # Robots are comfortable with integer names ;)
     id::Int32
 
     """
         Robot()
 
     Construct a robot with a unique name within its class.
     """
-    Robot() = new(mint_id())
+    Robot() = new(mint_id!())
```

Comment:
I didn't know that the trailing `!` is also used for functions that mutate global or module state, but I guess there are some functions that do it (like `Random.seed!`). Good to know :)

Reply:
I'm not actually 100% sure about this, but I think it makes sense.